Sources/SwiftInspectorAnalyzers/PropertyAnalyzer.swift

```diff
@@ -37,15 +37,11 @@ public final class PropertyAnalyzer: Analyzer {
 
   /// Finds property information for the provided type in the Swift file
   /// - Parameter fileURL: The fileURL where the Swift file is located
-  public func analyze(fileURL: URL) throws -> TypeProperties? {
+  public func analyze(fileURL: URL) throws -> Set<PropertyData>? {
     let syntax: SourceFileSyntax = try cachedSyntaxTree.syntaxTree(for: fileURL)
-    var result: TypeProperties?
-    let visitor = TypeSyntaxVisitor(typeName: typeName) { [unowned self] typeProperties in
```

Comment:
`Update type syntax visitor to modern approach` is a tricky commit. I do a number of things:
- I move the `merge(...)` method from `TypeProperties` to `TypeSyntaxVisitor`. It no longer throws.
- I write tests for the new `merge(...)` method.
- I update `TypeSyntaxVisitor` to the modern API of having a property of what it found rather than a callback.
- I remove `TypeProperties`.

After the commit the project builds and all tests pass if you comment out PropertyAnalyzerSpec.